### DISCUSSION TITLE
Change compiler version for Metal.

### DISCUSSION THREAD

**Discussion by immortalsalomon** [halide/Halide#8046] [Q&A]
Hello everyone, I hope you are well.

I created this discussion because when I try to run my compiled pipeline using Metal as target, the following error is generated.
![image](https://github.com/halide/Halide/assets/7589592/78fdfba6-503f-4b52-9665-63f0babf943e)

Searching online I found this issue: https://github.com/risc0/risc0/issues/999

My question is whether there is a way to define which compiler of Metal to use(version of Metal). And if the answer is yes, how can that be achieved.

Thank you for your support.

**Comment by shoaibkamil** [comment]
Can you possibly provide a reproducer?  We haven't been able to repro this locally.

I don't believe the changing the Metal language version will allow more or fewer temporary registers, so likely you need to specify a different schedule.  The linked issue seems to indicate there was an error that was fixed by a specific macOS version, not Metal compiler version.
